cmd2/cmd2.py

```diff
@@ -1794,16 +1723,16 @@ def complete(self, text, state):
                     try:
                         compfunc = getattr(self, 'complete_' + command)
                     except AttributeError:
-                        compfunc = self.completedefault
-
-                    subcommands = self.get_subcommands(command)
-                    if subcommands is not None:
-                        # Since there are subcommands, then try completing those if the cursor is in
-                        # the token at index 1, otherwise default to using compfunc
-                        index_dict = {1: subcommands}
-                        compfunc = functools.partial(self.index_based_complete,
-                                                     index_dict=index_dict,
-                                                     all_else=compfunc)
+                        # There's no completer function, next see if the command uses argparser
+                        cmd_func = getattr(self, 'do_' + command)
+                        if hasattr(cmd_func, 'has_parser') and hasattr(cmd_func, 'argparser') and \
+                                getattr(cmd_func, 'has_parser'):
+                            # Command uses argparser, switch to the default argparse completer
+                            argparser = getattr(cmd_func, 'argparser')
+                            compfunc = functools.partial(self._autocomplete_default,
+                                                         argparser=argparser)
+                        else:
+                            compfunc = self.completedefault
```

Comment:
Can ``has_parser`` ever be **False** if the command uses **argparse**? If not, can that attribute be removed and this code simplified to be:
```Python
try:
    compfunc = getattr(self, 'complete_' + command)
except AttributeError:
    # There's no completer function, next see if we should switch to the default argparse completer
    try:
        cmd_func = getattr(self, 'do_' + command)
        argparser = getattr(cmd_func, 'argparser')
        compfunc = functools.partial(self._autocomplete_default, argparser=argparser)
    except AttributeError:
        compfunc = self.completedefault
```